### PR TITLE
[CWS] merge common paths in activity dumps

### PR DESCRIPTION
**Pull request by paulcacheux**
### What does this PR do?

TODO:
- [x] add feature flag/config
- flag node, indicating that it contains a pattern
- white/black list folders ?
- proc heuristics
- folder merge
- patterns without prefix/suffix

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/utils/filemerge.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	prefix := make([]byte, 0)
+
+	a := ap.left
+	b := bp.left
+
+	for i := 0; i < len(a) && i < len(b) && a[i] < utf8.RuneSelf && a[i] == b[i]; i++ {
+		prefix = append(prefix, a[i])
+	}
+
+	return string(prefix)
```
Comment (lebauce) [inline pkg/security/utils/filemerge.go:54]: return string(a[:i) to avoid allocation ?

**Review by lebauce — COMMENTED** [review]
It seems that playing with pattern generates a tricky corner cases, it would be nice to have some unit tests

**Comment by paulcacheux** [comment]
Yes I'm working on more tests for activity dumps in general, to base the tests of this merge on

**Review by lebauce — APPROVED** [review]

**Review by sgnn7 — APPROVED** [review]
LGTM for `pkg/config/config.go` changes

**Review by Gui774ume — APPROVED** [review]

**Pull request merged**